pyiceberg/expressions/literals.py

```diff
@@ -588,7 +589,7 @@ def _(self, type_var: DecimalType) -> Literal[Decimal]:
     def _(self, type_var: BooleanType) -> Literal[bool]:
         value_upper = self.value.upper()
         if value_upper in ["TRUE", "FALSE"]:
-            return BooleanLiteral(value_upper == "TRUE")
+            return BooleanLiteral(strtobool(value_upper))
```

Comment:
I feel as though this is an over-optimization, since we check whether `value_upper` is in "TRUE" and "FALSE". WDYT?

An alternative is to instead just use strtobool for the whole function, but that would mean that we support a string expression like:

"col_A == 'off'" where col_A is a boolean column and 'off' gets parsed as False, and that feels unnecessarily over permissive. Should we leave this block of code as it is?

Reply:
Yeah, in this case, it makes sense to be strict.

Reply:
I reverted and added more tests to clarify that it needs to be more strict.

Reply:
Oh, my bad. I copied and pasted it but forgot to rerun the test.